src/components/forms/JurisdictionSelect/index.tsx

```diff
@@ -267,8 +269,22 @@ export const handleChangeWithOptions = (
           form.setFieldValue(field.name, optionVal.value);
           form.setFieldTouched(field.name, true);
           if (fiStatusFieldName) {
+            const jurisdictionIdx = field.name
+              .split('[')[1]
+              .split(']')[0]; /** get jurisdiction index from field name prop */
```

Comment:
What are we trying to do here?  I'm not sure I like how this looks and doubt if it will be not fragile.  We should take a step back and think if this can be solved/achieved in a different way.

Reply:
@moshthepitt The name prop of the field contains the index of jurisdiction, So we can get index from name prop and change its value.